### ISSUE TITLE
custom smileys do not work

### ISSUE THREAD

**Issue by nullinger** [bitnami/bitnami-docker-dokuwiki#40]
**Description**
[Custom smileys](https://www.dokuwiki.org/smileys) seem not to work out of the box

**Steps to reproduce the issue:**
1. Place image file under `lib/images/smileys/local/`
2. Create `conf/smileys.local.conf` and prefix each image filename with `local/…`

**Describe the results you received:**
Smileys do not appear

**Describe the results you expected:**
Smiley images can be chosen in the editor and appear on the pages.

**Additional information you deem important:**
Manually adding the folder as a separate docker volume works:
``` docker-compose.yml
    volumes:
      - './data:/bitnami'
      - './smileys/local:/opt/bitnami/dokuwiki/lib/images/smileys/local'
```

**Versions**
- docker: `18.02.0-ce`
-  docker-compose version: `1.8.0`
- DokuWiki: `Release 2017-02-19e "Frusterick Manners"`

**Comment by juan131** [comment]
I could not reproduce the issue. I was able to create a custom emoji following the docs you shared.

Could you please run the commands below and share the output?

```
docker-compose exec dokuwiki cat /opt/bitnami/dokuwiki/conf/smileys.local.conf
docker-compose exec dokuwiki ls -la /opt/bitnamidokuwiki/lib/images/smileys/local/
```

**Comment by nullinger** [comment]
```
$ docker-compose exec wiki cat /opt/bitnami/dokuwiki/conf/smileys.local.conf

# smileys.local.conf
:DOCKER:  local/docker.png
```
The directory containing the images does not seem to exist in the container, no matter if it is created manually in the mounted volume:
```
$ docker-compose exec wiki ls -la /opt/bitnami/dokuwiki/lib/images/smileys/local/
ls: cannot access /opt/bitnami/dokuwiki/lib/images/smileys/local/: No such file or directory
```
The parent folder ("smileys") does exist.

**Comment by juan131** [comment]
> The parent folder ("smileys") does exist.

That's the reason... I manually created the directory and download the `.jpg` file on it, using the command-line inside the container.

Anyway, I followed your approach and created a directory on my computer. Then, I downloaded a .png image on it. I called the directory "local" and then used it as a volume modifying the `docker-compose.yml`:

```
    volumes:
      - 'dokuwiki_data:/bitnami'
      - './local:/opt/bitnami/dokuwiki/lib/images/smileys/local'
```

As you can see, the `.png` file is where it should be after running `docker-compose up`:

```
$ docker-compose exec dokuwiki ls -la /opt/bitnami/dokuwiki/lib/images/smileys/local/
total 16
drwxr-xr-x 3 daemon daemon    96 Mar 13 09:22 .
drwxr-xr-x 3 daemon daemon  4096 Mar 13 09:24 ..
-rwxr-xr-x 1 daemon daemon 11423 Oct 31 15:18 docker.png
```

**Comment by nullinger** [comment]
> I manually created the directory and download the .jpg file on it, using the command-line inside the container.

My intention was not to change anything _inside_ the container because it will be lost when recreating it for example.

> Then, I downloaded a .png image on it. I called the directory "local" and then used it as a volume modifying the docker-compose.yml

This is a workaround that I already mentioned in my first post.

My expectation was to mount _one_ folder/volume from the host where _all_ persistent data is stored including custom smileys.
If that is not possible or wanted, this issue can be closed.

**Comment by juan131** [comment]
> My expectation was to mount one folder/volume from the host where all persistent data is stored including custom smileys.

That's what I did creating the "local" directory on the host and then mounting it on the container using:

``` 
    volumes:
      - 'dokuwiki_data:/bitnami'
      - './local:/opt/bitnami/dokuwiki/lib/images/smileys/local'
```

As you can see I'm mounting a folder that contains the custom smileys so they are persisted if the container dies.

**Comment by nullinger** [comment]
I think we are talking about different things.

Your solution is to use **_two_** volumes: The **data** folder and the **local** folder for the smileys.
This is exactly what I already mentioned in my first post. That works.

I just wanted to know if it would be possible to store the **local** folder inside the **data** folder so you have to mount only **_one_** volume that contains all the dokuwiki data.

**Comment by juan131** [comment]
Hi @nullinger 

Sorry for the misunderstanding... 

Current persisted directories on `/bitnami/dokuwiki` are: `["data", "conf", "lib/plugins", "lib/tpl"]`. You can check it by running:


```
ls -la /opt/bitnami/dokuwiki/data
lrwxrwxrwx 1 root root 22 Mar 15 10:20 /opt/bitnami/dokuwiki/data -> /bitnami/dokuwiki/data
ls -la /opt/bitnami/dokuwiki/conf
lrwxrwxrwx 1 root root 22 Mar 15 10:20 /opt/bitnami/dokuwiki/conf -> /bitnami/dokuwiki/conf
ls -la /opt/bitnami/dokuwiki/lib/plugins
lrwxrwxrwx 1 root root 29 Mar 15 10:20 /opt/bitnami/dokuwiki/lib/plugins -> /bitnami/dokuwiki/lib/plugins
ls -la /opt/bitnami/dokuwiki/lib/tpl
lrwxrwxrwx 1 root root 25 Mar 15 10:20 /opt/bitnami/dokuwiki/lib/tpl -> /bitnami/dokuwiki/lib/tpl
```

We can include "lib/images/smileys/local" to that list so they are persisted too. Would that work for you?

**Comment by nullinger** [comment]
Hi @juan131,

that seems to be the perfect solution.

Thank you.

**Comment by juan131** [comment]
Hi @nullinger 

We just released a new revision of the Dokuwiki container (https://github.com/bitnami/bitnami-docker-dokuwiki/releases/tag/0.20170219.201708232029-r5) including this directory to the dataToPersist. Could you please try using this latest version and let us know if that fixes your issue?

**Comment by nullinger** [comment]
Working for me now.
Thank you for your fast help and the good work!

**Comment by nullinger** [comment]
May I reopen this issue to also include Images for [InterWiki-Links](https://www.dokuwiki.org/InterWiki) (`lib/images/interwiki/`)?

Thank you

**Comment by nullinger** [comment]
Would be perfect  👍

**Comment by juan131** [comment]
Hi @nullinger I just released a new version of the Docker Image (0.20180422.201805030840-r5) which persists this directory. I also updated the `README.md` "Notable Changes" section reflecting this change. Please let us know if that fix your issue and feel free to close it once you consider it solved.

**Comment by nullinger** [comment]
Working for me now, thank you!
